backend/utils/Gemini.js

```diff
@@ -0,0 +1,31 @@
+const { GoogleGenerativeAI } = require("@google/generative-ai");
+
+// declare instructions for ai 
+const systemInstruction = "You are an AI designed to generate journal prompts for users based on their emotional states, which are influenced by their activities and mood. Ensure that the mood names and activity names in the initial prompt are included in the refined prompt. Users will not respond to these prompts. Your task is to refine the given initial prompt to make it more personal and relevant to the user's current emotions, and to ensure it aligns with the suggested tone included in the initial prompt. The tone should be woven naturally into the prompt to help the user think deeply about their feelings and experiences without expecting a reply. Your response should be a single, complete sentence crafted to help the user reflect on and better understand their emotions.";
```

Comment:
A couple pieces of feedback here: 
1. The sentence "users will not respond to these prompts" may be confusing to Gemini AI. It's unclear if you are referring to "the initial prompt" or "refined prompt." It might be good to clarify this. 
2. From the example journal prompt outputs you've shown me, it seems like most of the prompts don't refer back to the specific reason or trend in mood + activity that caused the prompt to be generated. For example, the prompt says something like "reflect on your out of place neutral reactions today", but ideally it could say something like "you are feeling neutral today - usually when you play basketball you are happy, so this feels like an outlier. Reflect on your out of place feelings today..." I think some simple clarifications in the prompt engineering could help improve this. Additionally, it feels like your prompt tells Gemini to "help the user reflect on...their emotions." Maybe if you say "help the user reflect on their emotion AND activities trends" it will generate prompts that are more likely to include specific activities that led the prompt to be generated. 

You can make this a stretch feature for TC #1 to implement in week 9. Focus on TC #2 for now! I just feel like your algorithm is so good and optimized, I want to make sure Gemini AI is actually using your algorithm to its highest potential! LMK if you want to discuss in person if this is confusing.

Reply:
I understand where you're coming from, but I believe this is covered. I designed the algorithm to detect unusual patterns in mood and activity, and then generate a prompt based on that. If there's nothing out of the ordinary, it'll simply ask about their last mood/ how they feel today.
I'm confident that this system is working as intended,  as I made sure to test Gemini in all cases, but I can see how my test plan video might not have covered every possible scenario. 
Also, I'll take your feedback on board and see if I can simplify the instructions I'm sending to Gemini.

Reply:
Thanks for clarifying - if you are confident with the prompts that are getting generated, I'm happy with this! No need to simplify it further.